src/fft-api/stock/fftStockService.ts

```diff
@@ -0,0 +1,353 @@
+import { Logger } from 'tslog';
+import { FftApiClient } from '../common';
+import {
+  FacilityServiceType,
+  FacilityStatus,
+  Stock,
+  StockDistribution,
+  StockForCreation,
+  StockForUpdate,
+  StockPaginatedResult,
+  StocksForUpsert,
+  StockSummaries,
+  StockUpsertOperationResult,
+} from '../types';
+import { CustomLogger } from '../../common';
+import { ResponseError } from 'superagent';
+
+export class FftStockService {
+  private readonly path: string = 'stocks';
+  private readonly logger: Logger<FftStockService> = new CustomLogger<FftStockService>();
+
+  constructor(private readonly apiClient: FftApiClient) {}
+
+  public async createStock(stockForCreation: StockForCreation): Promise<Stock> {
+    try {
+      return this.apiClient.post<Stock>(this.path, { ...stockForCreation });
+    } catch (error) {
+      const httpError = error as ResponseError;
+      this.logger.error(
+        `Could not create stock. Failed with status ${httpError.status}, error: ${
+          httpError.response ? JSON.stringify(httpError.response.body) : ''
+        }`
+      );
+
+      throw error;
+    }
+  }
+
+  public async getAll(
+    facilityRef?: string,
+    tenantArticleIds?: string[],
+    locationRefs?: string[],
+    size?: number,
+    startAfterId?: string
+  ): Promise<StockPaginatedResult> {
+    try {
+      const queryParams: Record<string, string> = {};
+
+      if (facilityRef) {
+        queryParams['facilityRef'] = facilityRef;
+      }
+
+      if (tenantArticleIds) {
+        tenantArticleIds = tenantArticleIds.slice(0, 499);
+
+        if (tenantArticleIds.length == 1) {
+          queryParams['tenantArticleIds'] = tenantArticleIds[0];
+        }
+
+        if (tenantArticleIds.length > 1) {
+          let query = `${tenantArticleIds[0]}&tenantArticleIds=`;
+          tenantArticleIds = tenantArticleIds.slice(1, 499);
+          query += tenantArticleIds.join('&tenantArticleIds=');
+
+          queryParams['tenantArticleIds'] = query;
+        }
+      }
+
```

Comment:
Maybe we could introduce a shared function for building the query so we can reuse it in multiple places.